site/source/_posts/2017/11/16-Dojo-2-Beta-4/Dojo-2-Beta-4.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Dojo 2 Beta 4
+date: 2017-11-16 16:56:19
+authorId: kitsonk
+featured_image: featured.png
+---
+
+The Beta 4 Release of Dojo 2 puts us on a clear path to releasing Dojo 2.  It has several significant changes and updates.
+
+{% asset_img featured.png feature-full %}
+
+<!-- more -->
+
+## Integrated virtual DOM
+
+This was the biggest fundamental change for the Beta 4 release.  We were really impressed with [Maquette](https://maquettejs.org/).  It was a focused, minimalistic, virtual DOM library.  We learned a lot from Maquette.  As we saw the roadmap for Maquette 3 materialise, we needed to re-evaluate our roadmap.  During the development of the widgeting system for Dojo 2, we really started to realise that _description_ of the virtual DOM and the _application_ were really linked.  We had created abstract objects, similar to the HyperScript nodes that Maquette used, but were specialised to our widgets and _standard_ DOM nodes.  We made the hard decision that we would continue to want features from a virtual DOM system that only be needed by Dojo 2, and therefore would unlikely to benefit a generic virtual DOM library.  We therefore made the hard decision to integrate the virtual DOM directly into `@dojo/widget-core`, where previously we had Maquette as a dependency.
+
+This alone was essentially transparent to end developers of Dojo 2, but it allowed us to address a whole list of features we desired and refactoring of code that introduced some significant **breaking changes** in Beta 4.
```

Comment:
I think transparent has the wrong context here. One of its definitions 
>having thoughts, feelings, or motives that are easily perceived.
might be misconstrued. Maybe "invisible" would better convey the message. Unless I'm just misunderstanding the message in the first place.